lib/swig/prox/prox_tv.i

```diff
@@ -14,14 +14,15 @@ class TProxTV : public TProx<T> {
              unsigned long start,
              unsigned long end,
              bool positive);
-};
 
-%template(ProxTV) TProxTV<double>;
-typedef TProxTV<double> ProxTV;
+  bool compare(const TProxTV<T> &that);
+};
 
 %template(ProxTVDouble) TProxTV<double>;
```

Comment:
Can you remind me when we can use `%template` and when we must use `%rename` ?

Reply:
In so far as I have found

template is fine if there is no complex types in parameter declarations - anywhere I have had to use "array<double> or array<float>" has failed with "array<T>" and so expansion using "rename" was found to work